official/transformer/compute_bleu.py

```diff
@@ -109,21 +111,22 @@ def main(unused_argv):
 
 
 if __name__ == "__main__":
-  parser = argparse.ArgumentParser()
-  parser.add_argument(
-      "--translation", "-t", type=str, default=None, required=True,
-      help="[default: %(default)s] File containing translated text.",
-      metavar="<T>")
-  parser.add_argument(
-      "--reference", "-r", type=str, default=None, required=True,
-      help="[default: %(default)s] File containing reference translation",
-      metavar="<R>")
-  parser.add_argument(
-      "--bleu_variant", "-bv", type=str, choices=["uncased", "cased"],
-      nargs="*", default=None,
-      help="Specify one or more BLEU variants to calculate (both are "
-           "calculated by default. Variants: \"cased\" or \"uncased\".",
-      metavar="<BV>")
-
-  FLAGS, unparsed = parser.parse_known_args()
-  main(sys.argv)
+  flags.DEFINE_string(
+      name="translation", short_name="t", default=None,
```

Comment:
The collision of abbreviations is inevitable at this point. Maybe we should make a rule that we don't have abbreviations for flags defined within model modules. Else, we might add a flag to the base set, conflict, and not realize it until someone tries to run an inheriting model and it either errors out at arg load time, or treats flags in strange ways. Thoughts, @robieta ?

Reply:
I think at this point one letter abbreviations are too dangerous to add, because they may collide not only with our own flags but flags defined who knows where. Two letters seems reasonably safe. 

If we're doing an end-to-end test it will blow up for collisions, so I'm less worried that we will accidentally have internal collisions.

Reply:
Right now mnist is missing it. It requires synthetic to be hermetic which I think is why mnist doesn't have it yet.

I definitely think it is something we want for every model.